tests/helpers_tests/test_condition.py

```diff
@@ -371,6 +429,42 @@ def test_greater_than_int__returns_true__when_user_value_greater_than_condition_
 
       self.assertStrictTrue(evaluator.evaluate(0))
 
+  def test_greater_than_int__calls_is_finite_number(self):
+    """ Returns True if is_finite_number returns True. Returns None if is_finite_number returns False. """
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      gt_int_condition_list, {'meters_travelled': 48.1}
+    )
+
+    with mock.patch('optimizely.helpers.validator.is_finite_number',
+                    return_value=True) as is_finite:
+      self.assertStrictTrue(evaluator.evaluate(0))
+    is_finite.assert_called_with(48.1)
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      gt_int_condition_list, {'meters_travelled': 49}
+    )
+
+    def side_effect(*args):
+      if args[0] == 49:
+        return False
+      return True
+
+    with mock.patch('optimizely.helpers.validator.is_finite_number',
+                    side_effect=side_effect) as is_finite:
+      self.assertIsNone(evaluator.evaluate(0))
+    is_finite.assert_any_call(49)
+
+    if PY2:
+      evaluator = condition_helper.CustomAttributeConditionEvaluator(
+        gt_int_condition_list, {'meters_travelled': long(49)}
+      )
+
+      with mock.patch('optimizely.helpers.validator.is_finite_number',
+                       side_effect=side_effect) as is_finite:
+        self.assertIsNone(evaluator.evaluate(0))
+      is_finite.assert_called_with(long(49))
+
```

Comment:
Ooh, I like the idea of using `side_effect` here. Can we go farther with this and test that `is_finite` only needs to reject the condition value _or_ the user attribute value in order to stop condition evaluation?

Something like
```python
def test_greater_than__calls_is_finite_number(self):
  evaluator = condition_helper.CustomAttributeConditionEvaluator(
    gt_int_condition_list, {'meters_travelled': 48.1}
  )
  
  def is_finite_number__rejecting_condition_value(value):
    if value == 48:
      return False
    return True
  
  with mock.patch('optimizely.helpers.validator.is_finite_number',
                  side_effect=is_finite_number__rejecting_condition_value):
    self.assertIsNone(evaluator.evaluate(0))
  
  def is_finite_number__rejecting_user_attribute_value(value):
    if value == 48.1:
      return False
    return True
  
  with mock.patch('optimizely.helpers.validator.is_finite_number',
                  side_effect=is_finite_number__rejecting_user_attribute_value):
    self.assertIsNone(evaluator.evaluate(0))
  
  def is_finite_number__accepting_both_values(value):
    return True
  
  with mock.patch('optimizely.helpers.validator.is_finite_number',
                  side_effect=is_finite_number__accepting_both_values):
    self.assertIsNone(evaluator.evaluate(0))
```

I don't think we need to test ints, longs, and floats separately... not here, at least... we can probably get away with `test_greater_than__calls_is_finite_number`, `test_less_than__calls_is_finite_number`, and maybe ``test_exact__given_number_values__calls_is_finite_number`. If that makes sense.

@oakbani @aliabbasrizvi what do you think?

Reply:
Nice suggestion. Revised accordingly.